File: backend/services/budget-service/internal/domain/budget.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type BudgetPeriod string

const (
	BudgetPeriodMonthly BudgetPeriod = "monthly"
	BudgetPeriodWeekly  BudgetPeriod = "weekly"
	BudgetPeriodCustom  BudgetPeriod = "custom"
)

type Budget struct {
	ID              uuid.UUID
	UserID          uuid.UUID
	Name            string
	PeriodType      BudgetPeriod
	CategoryID      *uuid.UUID
	StartDate       time.Time
	EndDate         time.Time
	LimitAmountMinor int64
	Currency        string
	Version         int64
	CreatedAt       time.Time
	UpdatedAt       time.Time
	DeletedAt       *time.Time
}

File: mobile/velune/lib/domain/models/budget.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'budget.freezed.dart';
part 'budget.g.dart';

enum BudgetPeriodType {
  @JsonValue('monthly')
  monthly,
  @JsonValue('weekly')
  weekly,
  @JsonValue('custom')
  custom,
}

@freezed
class Budget with _$Budget {
  const factory Budget({
    required String id,
    required String userId,
    required String name,
    required BudgetPeriodType periodType,
    String? categoryId,
    required String startDate,
    required String endDate,
    required int limitAmountMinor,
    required String currency,
    required int version,
  }) = _Budget;

  factory Budget.fromJson(Map<String, dynamic> json) =>
      _$BudgetFromJson(json);
}

